test/cpp/api/nn_utils.cpp

```diff
@@ -41,7 +41,7 @@ TEST_F(NNUtilsTest, ClipGradNorm) {
     for (int i = 0; i < grads.size(); i++) {
       auto param = l->parameters()[i];
       auto grad = grads[i];
-      p_scale.push_back(param.grad().data().div(grad).view(-1));
+      p_scale.push_back(param.mutable_grad().data().div(grad).view(-1));
```

Comment:
So this one is a read-only use of grad right?

Reply:
Yes my bad.